src/cdk/coercion/element.ts

```diff
@@ -0,0 +1,17 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {ElementRef} from '@angular/core';
+
+/**
+ * Coerces an ElementRef or an Element into an element.
+ * Useful for APIs that can accept either a ref or the native element itself.
+ */
+export function coerceElement<T>(elementOrRef: ElementRef<T> | T): T {
+  return elementOrRef instanceof ElementRef ? elementOrRef.nativeElement : elementOrRef;
```

Comment:
Maybe a pointless micro-optimization, but generally checking for the presence of a property is faster than `instanceof`. I think you could do
```ts
return elementOrRef.nativeElement || elementOrRef;
```

Reply:
TypeScript will complain about that one since `T` doesn't always have a `nativeElement` property.

Reply:
Yeah, I'd be okay with using type coercion here. I don't feel super strongly about it, though.